### PR TITLE
abci: minor cleanups in the socket client

### LINKED ISSUE #3513: abci: minor cleanups in the socket client
Follow up from https://github.com/tendermint/tendermint/pull/3512

Specifically:
- `cli.conn.Close()` need not be under the mutex (https://github.com/tendermint/tendermint/pull/3512#discussion_r270639920)
- call the reqRes callback after the resCb so they always happen in the same order (https://github.com/tendermint/tendermint/pull/3512/#discussion_r270640328)

### PR DESCRIPTION
**Pull request by melekes**
Follow up from #3512

Specifically:

    cli.conn.Close() need not be under the mutex (#3512 (comment))
    call the reqRes callback after the resCb so they always happen in the same order (#3512)

Fixes #3513

* [x] Referenced an issue explaining the need for the change
* [ ] ~~Updated all relevant documentation in docs~~
* [x] Updated all code comments where relevant
* [ ] ~~Wrote tests~~
* [ ] ~~Updated CHANGELOG_PENDING.md~~

### CODE REVIEW

File: abci/client/socket_client.go [hunk 1]
```diff
@@ -82,15 +78,12 @@ RETRY_LOOP:
 }
 
 func (cli *socketClient) OnStop() {
-	cli.BaseService.OnStop()
```
Comment (marbar3778) [inline abci/client/socket_client.go]: should we also remove this function and `BaseService.OnStart()` as they don't do anything, then they can be removed elsewhere?

Comment (melekes) [reply abci/client/socket_client.go]: These functions act as defaults and sometimes overwritten (e.g. in BaseReactor). But we can safely remove BaseService.OnStart and OnStop calls (what I did here).

Comment (marbar3778) [reply abci/client/socket_client.go]: 👍 ah okay

**Comment by melekes** [comment]
> Also, does this warrant a (pending) change log entry?

I'd say no

**Comment by ebuchman** [comment]
> > Also, does this warrant a (pending) change log entry?
> 
> I'd say no

The re-ordering change here might be significant enough to warrant a changelog entry

**Pull request merged**